classes/ZipAction.php

```diff
@@ -165,21 +165,37 @@ public function extract(string $from_file, string $to_dir): bool
         }
 
         for ($i = 0; $i < $zip->numFiles; ++$i) {
-            if (!$zip->extractTo($to_dir, [$zip->getNameIndex($i)])) {
-                $error = $zip->getStatusString();
+            $fileName = $zip->getNameIndex($i);
+            if (!$zip->extractTo($to_dir, [$fileName])) {
+                $stat = $zip->statIndex($i);
+                $filePath = $stat['name'];
+
+                $fullSystemPath = rtrim($to_dir, DIRECTORY_SEPARATOR) . DIRECTORY_SEPARATOR . $filePath;
+                $md5FromZip = md5($zip->getFromIndex($i));
+                $md5FromSystem = is_file($fullSystemPath) ? md5_file($fullSystemPath) : null;
```

Comment:
```suggestion
                $md5FromDisk = is_file($fullSystemPath) ? md5_file($fullSystemPath) : null;
```

`System` makes me feed we are talking about the System32 folder of Windows, or another folder of an operating software.